protostar/starknet_gateway/gateway_facade.py

```diff
@@ -94,15 +95,18 @@ async def deploy(
             },
         )
 
-        result = await self._gateway_client.deploy(tx, token)
-        register_response(dataclasses.asdict(result))
-        if wait_for_acceptance:
-            if self._logger:
-                self._logger.info("Waiting for acceptance...")
-            _, status = await self._gateway_client.wait_for_tx(
-                result.transaction_hash, wait_for_accept=wait_for_acceptance
-            )
-            result.code = status
+        try:
+            result = await self._gateway_client.deploy(tx, token)
+            register_response(dataclasses.asdict(result))
+            if wait_for_acceptance:
+                if self._logger:
+                    self._logger.info("Waiting for acceptance...")
+                _, status = await self._gateway_client.wait_for_tx(
+                    result.transaction_hash, wait_for_accept=wait_for_acceptance
+                )
+                result.code = status
+        except TransactionFailedError as ex:
+            raise TransactionException(str(ex)) from None
```

Comment:
```suggestion
            raise TransactionException(str(ex)) from ex
```

etc.

`from None` part has no sense to me and I would rather keep causality chain

Reply:
done